### PR TITLE
fix(funnels): Validate `funnelStep`

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Sentry posted about issue [POSTHOG-1D0B](https://posthog.sentry.io/issues/5538141641/) on https://github.com/PostHog/posthog/pull/23135#issuecomment-2190921450. It looks like the timing of that PR and the issue popping up is just accidental, and I didn't find one that touched `funnelStep` recently, so not sure how a couple of users got into this situation. But validation should catch this earlier.

## Changes

Added guard against `funnelStep` being 0 or -1.

## How did you test this code?

Uncommented and enhanced a test against this problem.

### CODE REVIEW

File: posthog/hogql_queries/insights/funnels/base.py [hunk 1]
```diff
@@ -315,6 +315,25 @@ def _serialize_step(
...
         return self._extra_event_fields + self._extra_event_properties
 
+    @property
+    def _absolute_actors_step(self) -> Optional[int]:
+        """The actor query's 1-indexed target step converted to our 0-indexed SQL form. Never a negative integer."""
+        if self.context.actorsQuery is None:
+            return None
+        target_step = self.context.actorsQuery.funnelStep
+        if target_step is None:
+            return None
```
Comment (thmsobrmlr) [inline posthog/hogql_queries/insights/funnels/base.py:325]:
```suggestion
        if self.context.actorsQuery is None or self.context.actorsQuery.funnelStep is None:
            return None

        target_step = self.context.actorsQuery.funnelStep
```
nittiest of nit-picks, but somehow I find this more readable

File: posthog/hogql_queries/insights/funnels/base.py [hunk 2]
```diff
@@ -315,6 +315,25 @@ def _serialize_step(
...
+        if target_step < 0:
+            # the first valid dropoff argument for funnel_step is -2
+            # -2 refers to persons who performed the first step but never made it to the second
+            if target_step == -1:
+                raise ValueError("To request dropoff of initial step use -2")
```
Comment (thmsobrmlr) [inline posthog/hogql_queries/insights/funnels/base.py:330]:
```suggestion
            if target_step == -1:
                raise ValueError("The first valid drop-off argument for funnelStep is -2. -2 refers to persons who performed the first step but never made it to the second.")
```
I'd honestly just put the comment into the error message, as it's easier to understand for someone without context.

**Review by thmsobrmlr — APPROVED** [review]
Great, thank you! Two suggestions inline - do with them as you please.

**Comment by Twixes** [comment]
Good suggestions, applied @thmsobrmlr

**Review by skoob13 — APPROVED** [review]

**Pull request merged**